doc/api/stream.md

```diff
@@ -1799,6 +1799,32 @@ Calling `Readable.from(string)` or `Readable.from(buffer)` will not have
 the strings or buffers be iterated to match the other streams semantics
 for performance reasons.
 
+### `stream.addAbortSignal(signal, readable)`
+<!-- YAML
+added: REPLACEME
+-->
+* `signal` {AbortSignal} A signal representing possible cancellation
+* `readable` {sream.Readable} a readable stream to attach the signal to
+
+Attaches an AbortSignal to a readable stream. This lets code control stream
+destruction using an `AbortController`.
+
+Calling `abort` on the `AbortController` corresponding to the passed
+`AbortSignal` will behave the same way as calling `.destroy(new AbortError())`
+on the stream.
+
+```js
+const fs = require('fs');
+
+const controller = new AbortController();
+const read = addAbortSignal(
+  controller.signal,
+  fs.createReadStream(('object.json'))
+);
+// Later, abort the operation closing the stream
+controller.abort();
+```
```

Comment:
An example that shows this used with an async iterator would be good.

Reply:
I really don’t like this. Would much prefer a signal constructor argument.